server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -827,15 +840,22 @@ public Iterable<String> modifySecretGroups(@Auth AutomationClient automationClie
   @DELETE
   @Path("{name}")
   @LogArguments
-  public Response deleteSecretSeries(@Auth AutomationClient automationClient,
-      @PathParam("name") String name) {
+  public Response deleteSecretSeries(
+      @Auth AutomationClient automationClient,
+      @PathParam("name") String name,
+      @QueryParam("deletionMode") String deletionMode) {
+
+    SecretDeletionMode mode = deletionMode == null
+        ? SecretDeletionMode.SOFT
+        : EnumUtils.getEnumIgnoreCase(SecretDeletionMode.class, deletionMode);
```

Comment:
In the client-facing API equivalent of this enum I used lowercase constants (since I imagined them being dumped directly into URLs as query parameters) but I followed the Java naming standard of all-caps for the actual Java enum. Since enum lookups are case-sensitive (gross), this Commons utility method prevents any case-related mixups.